catalog/core/catalog-core-impl/filter-proxy/src/main/java/ddf/catalog/filter/proxy/adapter/GeotoolsFilterAdapterImpl.java

```diff
@@ -103,14 +103,22 @@
 
   public static final String CQL_KILOMETERS = "kilometers";
 
+  private static final String EXPRESSION_NOT_SUPPORTED =
+      " expression not supported by Filter Adapter.";
+
+  private static final String FILTER_NOT_SUPPORTED = " filter not supported by Filter Adapter.";
+
   private static final double ERROR_THRESHOLD = .000001;
 
   private static final FilterFactory FF = new FilterFactoryImpl();
 
-  // Does not support fractional values
-  private static final Pattern RELATIVE_TEMPORAL_MATCHER =
-      Pattern.compile(
-          "RELATIVE\\(P(?!$)(\\d+Y)?(\\d+M)?(\\d+W)?(\\d+D)?(T(?=\\d)(\\d+H)?(\\d+M)?(\\d+S)?)?\\)");
+  private static final String DECIMAL_REGEX = "\\\\d*\\\\.\\\\d+|\\\\d+\\\\.?\\\\d*";
+
+  private static final String SHORTENED_RELATIVE_TEMPORAL_REGEX =
+      "RELATIVE\\(P(?!$)(?:(dec)Y)?(?:(dec)M)?(?:(dec)W)?(?:(dec)D)?(?:T(?=dec)(?:(dec)H)?(?:(dec)M)?(?:(dec)S)?)?\\)";
```

Comment:
I'm reading online and it looks like decimals are only supposed to be supported for the lowest order date and the lowest order time. In other words, if there's a days present, you can't have a decimal on the months, weeks, or years. Same applies to time

Reply:
Yea according to the ISO spec only the seconds field should contain a decimal in a period like this, however, my understanding is the UI component team discussed this and decided they wanted to do unit conversion when the user inputs a decimal value, rather than just deny it. @andrewkfiedler can you chime in here?

Reply:
I think the minutes can contain a decimal if it's the lowest order unit specified in the Period string we hand it. I can't remember offhand if we shorten something like 1 year and 1 month to 1y1m or if we fill it in like 1y1m0w0d (what I wrote probably isn't ISO). But I think if we're shortening it then in this case the month can have a decimal.

I think if we really want to support being non-ISO compliant, are we able to update it on the fly in the UI?

Reply:
The UI will always send over compliant ISO as is.  It only allows one unit at a time for a period, so decimals are always fine.

Reply:
Just to double check, is a UI search the only path that will hit this code?

Reply:
I want to say yes just because the regex is called RELATIVE_TEMPORAL_REGEX, But I can't be sure and I don't know how to check.

Reply:
nvm you can hit it using a cql search from the console so I'm sure there are lot's of other ways to get to it.